src/common/components/Transform.hpp

```diff
@@ -0,0 +1,78 @@
+/*
+** EPITECH PROJECT, 2024
+** StellarForge
+** File description:
+** No file there , just an epitech header example .
+*/
+
+#ifndef STELLARFORGE_TRANSFORM_HPP
+#define STELLARFORGE_TRANSFORM_HPP
+
+#include "../../physics/Physics.hpp"
+#include "../IComponent.hpp"
+#include "../fields/Vector3Field.hpp"
+#include "../fields/FieldGroup.hpp"
+
+using Vector3 = glm::vec3;
+
+class Transform : virtual public IComponent {
+private:
+    Vector3 position;
+    glm::quat rotation;
+    Vector3 scale;
+
+public:
+    Transform() : position(0, 0, 0), rotation(0, 0, 0, 1), scale(1, 1, 1) {}
```

Comment:
Missing the details for each function prototype is like :
```            
/**
 * @brief "exemple"
 * @param "exemple"
 * @param "exemple"
 * @version 0.1.0
 * @since 0.1.0
 * @author xxxx
 */
```